sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Implicits.scala

```diff
@@ -156,15 +159,17 @@ private[sql] object CatalogV2Implicits {
 
     def asTableIdentifier: TableIdentifier = parts match {
       case Seq(tblName) => TableIdentifier(tblName)
-      case Seq(dbName, tblName) => TableIdentifier(tblName, Some(dbName))
+      case Seq(dbName, tblName) =>
+        TableIdentifier(tblName, Some(dbName), sessionCatalogOption(dbName))
```

Comment:
how about `MultipartIdentifierHelper`? Why it's useful to attach catalog name here?

Reply:
it also be used during resolve v2 -> v1 command, e.g. `CreateTable#ResolvedDBObjectName`, if its catalog is session catalog, we will convert the multipart name to TableIdentifier then to construct v1 CreateTable.